precompile/contracts/rewardmanager/simulated_test.go

```diff
@@ -0,0 +1,322 @@
+				testContractAddr, testContract := deployRewardManagerTest(t, backend, admin)
+				allowlisttest.SetAsEnabled(t, backend, rewardManager, admin, testContractAddr)
+
+				isAllowed, err := testContract.AreFeeRecipientsAllowed(nil)
+				require.NoError(t, err)
+				require.False(t, isAllowed)
+			},
+		},
+		{
+			name: "should allow enabled contract to allow fee recipients",
+			test: func(t *testing.T, backend *sim.Backend, rewardManager *rewardmanagerbindings.IRewardManager) {
+				testContractAddr, testContract := deployRewardManagerTest(t, backend, admin)
+				allowlisttest.SetAsEnabled(t, backend, rewardManager, admin, testContractAddr)
+
+				tx, err := testContract.AllowFeeRecipients(admin)
+				require.NoError(t, err)
+				testutils.WaitReceiptSuccessful(t, backend, tx)
+
+				isAllowed, err := testContract.AreFeeRecipientsAllowed(nil)
+				require.NoError(t, err)
+				require.True(t, isAllowed)
+			},
+		},
+		{
+			name: "should allow enabled contract to disable rewards",
+			test: func(t *testing.T, backend *sim.Backend, rewardManager *rewardmanagerbindings.IRewardManager) {
+				testContractAddr, testContract := deployRewardManagerTest(t, backend, admin)
+				allowlisttest.SetAsEnabled(t, backend, rewardManager, admin, testContractAddr)
+
+				tx, err := testContract.SetRewardAddress(admin, testContractAddr)
+				require.NoError(t, err)
+				testutils.WaitReceiptSuccessful(t, backend, tx)
+
+				currentAddr, err := testContract.CurrentRewardAddress(nil)
+				require.NoError(t, err)
+				require.Equal(t, testContractAddr, currentAddr)
+
+				tx, err = testContract.DisableRewards(admin)
+				require.NoError(t, err)
+				testutils.WaitReceiptSuccessful(t, backend, tx)
+
+				currentAddr, err = testContract.CurrentRewardAddress(nil)
+				require.NoError(t, err)
+				require.Equal(t, constants.BlackholeAddr, currentAddr)
+			},
+		},
+		{
+			name: "should return blackhole as default reward address",
+			test: func(t *testing.T, _ *sim.Backend, rewardManager *rewardmanagerbindings.IRewardManager) {
+				currentAddr, err := rewardManager.CurrentRewardAddress(nil)
+				require.NoError(t, err)
+				require.Equal(t, constants.BlackholeAddr, currentAddr)
+			},
+		},
+		{
+			name: "fees should go to blackhole by default",
+			test: func(t *testing.T, backend *sim.Backend, _ *rewardmanagerbindings.IRewardManager) {
+				client := backend.Client()
+
+				initialBlackholeBalance, err := client.BalanceAt(t.Context(), constants.BlackholeAddr, nil)
+				require.NoError(t, err)
+
+				tx := testutils.SendSimpleTx(t, backend, adminKey)
+				testutils.WaitReceiptSuccessful(t, backend, tx)
+
+				newBlackholeBalance, err := client.BalanceAt(t.Context(), constants.BlackholeAddr, nil)
+				require.NoError(t, err)
+
+				require.Positive(t, newBlackholeBalance.Cmp(initialBlackholeBalance),
+					"blackhole balance should have increased from fees")
+			},
+		},
+		{
+			name: "fees should go to configured reward address",
```

Comment:
we can also test if the precompile allows configured coinbase/etherbase with in environment (a.k.a `allowFeeRecipients`. We can add a option for `WithEtherbase`  for configuring the `Etherbase` [here](https://github.com/ava-labs/subnet-evm/blob/master/ethclient/simulated/backend.go#L110) (similar to https://github.com/ava-labs/subnet-evm/blob/master/ethclient/simulated/options.go#L56). So we can have a backed running with a different coinbase (!= blackhole) to see if the fees are directed to that custom coinbase. It might require a precompile config that enabled the `allowFeeRecipients` in genesis through [this.](https://github.com/ava-labs/subnet-evm/blob/master/precompile/contracts/rewardmanager/config.go)